examples/helloworld/client.ts

```diff
@@ -496,4 +499,123 @@ export class HelloWorldAppClient {
+        promiseChain = promiseChain.then(() => client.clearState({...args, sendParams: {...args?.sendParams, skipSending: true, atc}}))
+        resultMappers.push(undefined)
+        return this
+      },
+      async execute() {
+        await promiseChain
+        const result = await algokit.sendAtomicTransactionComposer({ atc, sendParams: {} }, client.algod)
+        return {
+          ...result,
+          returns: result.returns?.map((val, i) => val.returnValue !== undefined ? resultMappers[i]?.(val.returnValue) : undefined)
+        }
+      }
+    } as unknown as HelloWorldAppComposer
+  }
+}
+export type HelloWorldAppComposer<TSignatures extends [...(undefined | HelloWorldAppSig)[]] = []> = {
+  /**
+   * Calls the hello(string)string ABI method.
+   *
+   * Returns Hello, {name}
+   *
+   * @param args The arguments for the contract call
+   * @param params Any additional parameters for the call
+   * @returns The result of the call
+   */
+  hello(args: MethodArgs<'hello(string)string'>, params?: AppClientCallCoreParams & CoreAppCallArgs): HelloWorldAppComposer<[...TSignatures, 'hello(string)string']>
+
+  /**
+   * Calls the hello_world_check(string)void ABI method.
+   *
+   * Asserts {name} is "World"
+   *
+   * @param args The arguments for the contract call
+   * @param params Any additional parameters for the call
+   * @returns The result of the call
+   */
+  helloWorldCheck(args: MethodArgs<'hello_world_check(string)void'>, params?: AppClientCallCoreParams & CoreAppCallArgs): HelloWorldAppComposer<[...TSignatures, 'hello_world_check(string)void']>
+
+  /**
+   * Gets available update methods
+   */
+  readonly update: {
+    /**
+     * Updates an existing instance of the HelloWorldApp smart contract using a bare call.
+     *
+     * @param args The arguments for the bare call
+     * @returns The update result
+     */
+    bare(args?: BareCallArgs & AppClientCallCoreParams & AppClientCompilationParams & CoreAppCallArgs): HelloWorldAppComposer<[...TSignatures, undefined]>
+  }
+
+  /**
+   * Gets available delete methods
+   */
+  readonly delete: {
+    /**
+     * Deletes an existing instance of the HelloWorldApp smart contract using a bare call.
+     *
+     * @param args The arguments for the bare call
+     * @returns The delete result
+     */
+    bare(args?: BareCallArgs & AppClientCallCoreParams & CoreAppCallArgs): HelloWorldAppComposer<[...TSignatures, undefined]>
+  }
+
+  /**
+   * Makes a clear_state call to an existing instance of the HelloWorldApp smart contract.
+   *
+   * @param args The arguments for the bare call
+   * @returns The clear_state result
+   */
+  clearState(args?: BareCallArgs & AppClientCallCoreParams & CoreAppCallArgs): HelloWorldAppComposer<[...TSignatures, undefined]>
+
+  /**
+   * Executes the transaction group and returns an array of results
+   */
+  execute(): Promise<HelloWorldAppComposerResults<TSignatures>>
```

Comment:
I think it would also be handy to have a build method that returns the raw transaction objects too?

Reply:
Similarly, there should be an addTransaction escape hatch